### PR TITLE
Fixed #27793 -- Leveraged new ipaddress module to validate IP addresses

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

**Comment by aaugustin** [comment]
Good job, removing over 200 lines of code there 👍

File: django/utils/ipv6.py [hunk 1]
```diff
@@ -10,139 +7,32 @@
...
 
     Args:
         ip_str: A valid IPv6 address.
         unpack_ipv4: if an IPv4-mapped address is found,
         return the plain IPv4 address (default=False).
         error_message: An error message used in the ValidationError.
 
-    Returns:
-        A compressed IPv6 address, or the same value
+    Return a compressed IPv6 address, or the same value.
```
Comment (timgraham) [inline django/utils/ipv6.py]: The comma isn't needed.

File: django/utils/ipv6.py [hunk 2]
```diff
@@ -10,139 +7,32 @@
...
-    Cleans an IPv6 address string.
+    Clean an IPv6 address string.
 
-    Validity is checked by calling is_valid_ipv6_address() - if an
-    invalid address is passed, ValidationError is raised.
+    If an invalid address is passed, ValidationError is raised.
 
-    Replaces the longest continuous zero-sequence with "::" and
-    removes leading zeroes and makes sure all hextets are lowercase.
+    Replace the longest continuous zero-sequence with "::", remove leading
```
Comment (timgraham) [inline django/utils/ipv6.py]: I guess the `ipaddress` module does these things? I'm not sure if it's behavior should be reiterated in detail.

Comment (claudep) [inline django/utils/ipv6.py]: I'll commit that for now, but feel free to simplify things if you think it's better.

**Pull request closed without merging**